src/test/java/com/salesforce/dynamodbv2/grammar/ExpressionsTest.java

```diff
@@ -101,14 +101,14 @@ void testParseUpdateExpression(String expression) {
 
         // assert both trees match
         int level = 0;
-        for (String ourTreeLevel: ourTreeLevels) {
+        for (String ourTreeLevel : ourTreeLevels) {
             String dynamodbTreeLevel = dynamodbTreeLevels.get(level);
 
             if (dynamodbTreeLevel.contains("_section") || dynamodbTreeLevel.contains("_action")
                 || dynamodbTreeLevel.contains("_value")) {
                 // dynamodb and our tree contain the same text but differ in the delimiter used and case sensitivity
-                dynamodbTreeLevel.replace("_", "").toLowerCase();
-                ourTreeLevel.toLowerCase();
+                assertEquals(dynamodbTreeLevel.replace("_", "").toLowerCase().trim(),
```

Comment:
before the change, modifications of `dynamodbTreeLevel` and `ourTreeLevel` were being computed but not tested (unlike in the other (`else if` and `else`) branches)

Reply:
yes the assertion should be there, thanks